applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -253,6 +295,24 @@ static bool app_event_handler(const struct app_event_header *aeh)
 		return false;
 	}
 
+	if (IS_ENABLED(CONFIG_CAF_BLE_SMP_TRANSFER_EVENTS) && is_ble_smp_transfer_event(aeh)) {
+		process_dvfs_states(DVFS_STATE_SMP_TRANSFER, true);
+		k_work_reschedule(
+		 &dvfs_state_timeout_data[DVFS_STATE_SMP_TRANSFER].dvfs_timeout_work,
+		 K_MSEC(dvfs_state_timeout_data[DVFS_STATE_SMP_TRANSFER].dfvs_state_timeout_ms));
+
+		return false;
+	}
+
+	if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE) && is_config_event(aeh)) {
+		process_dvfs_states(DVFS_STATE_CONFIG_CHANNEL, true);
+		k_work_reschedule(
```

Comment:
You can easily filter out DVFS states that do not use timeout, because value of configured timeout for them is zero. Moving reschedule to `process_dvfs_states` would result in more common code and it would reduce chance that work would not be rescheduled when needed (code would be less error prone).

Reply:
I don't like your idea as I can imagine cases that someone might want to use timeout for a state that normally uses event to go to LOW_FREQ. 

We can discuss whether "code would be less error prone" vs " to show explicitly that we reschedule timeout."  is better and have 3 arguments or leave it as it is.

Reply:
@alstrzebonski what do you think?

Reply:
If you both prefer the existing approach I would not insist (behavior is proper anyway)